R/likelihood.R

```diff
@@ -77,9 +126,25 @@ likelihood <- function(chains, statistic = c("size", "length"), offspring_dist,
   checkmate::assert_numeric(
     exclude, null.ok = TRUE
   )
+  # likelihood can also work with a summarised <epichains_tree> object
+  if (.is_epichains(chains)) {
+    chains <- summary(chains)
+  }
+
+  if (any(chains == stat_max) && is.infinite(stat_max)) {
```

Comment:
Nice! I agree with reimplementing it as S3, setting the default value of `stat_max` to `NULL`, and checking that `stat_max` is specified (using `missing()`).

The major issue here is where we have to do (assuming a user specifies `stat_max = Inf`) https://github.com/epiverse-trace/epichains/blob/b090f4c51da700eb5da237928e5e022435fb1e06/R/likelihood.R#L108. If `stat_max` is `Inf`, this will error since R can't generate the sequence.

Reply:
> Nice! I agree with reimplementing it as S3, setting the default value of `stat_max` to `NULL`, and checking that `stat_max` is specified (using `missing()`).
> 
> The major issue here is where we have to do (assuming a user specifies `stat_max = Inf`)
> 
> https://github.com/epiverse-trace/epichains/blob/b090f4c51da700eb5da237928e5e022435fb1e06/R/likelihood.R#L108
> 
> . If `stat_max` is `Inf`, this will error since R can't generate the sequence.

To try to solve this, I've re-read the [Farrington paper](https://academic.oup.com/biostatistics/article/4/2/279/245106), especially Section 2 on Unconditional likelihoods (+ section on censoring) and come up with a few ideas/assumptions.

1. If the data supplied is numeric (not `<epichains>` or `<epichains_summary>`), then the user has control over censoring scheme.
2. If the stata contains `Inf` and `stat_max` is specified as `Inf`, `likelihood()` will error and request that the data be recensored with arbitrarily large (but realistic) finite values.

Reply:
From meeting today:

Alternatively, the error could say the vector should not contain infinite values, so replace with finite censoring values.